### ISSUE TITLE
Bug when building NAMESPACE for S4 Classes

### ISSUE THREAD

**Issue by ChristopherEeles** [r-lib/roxygen2#1200]
Hello roxygen2 team,

I believe I have found a bug in the roxygen2 package for R devel. The bug causes NAMESPACE and manual generation to fail for packages using S4 classes. This does not occur if I switch back to R 4.0.3,

The error is as follows:
```
First time using roxygen2. Upgrading automatically...
Loading testPkg
Error in add_classes_to_exports(ns = nsenv, package = package, exports = exports,  : 
  object 'lev' not found
```
For stepping through the `roxygen2::roxygenise` function I determined that `lev` is the name of a variable in one of the roxygenise helper methods.

Minimal reproduction:
- Make a package with `usethis::create_package('testPkg')`
- Add an R file with a dummy S4 class, in my case I made `dummyClass-class.R` with:
```
#' dummyClass
#'
#' @export
dummyClass <- setClass('dummyClass', contains='list')
```
- Try to roxygenise: `roxygen2::roxygenise()`

If you could help resolve this it would be appreciated, as it prevent me from roxygenising my Bioconductor package in R devel.

sessionInfo:
```
 Under development (unstable) (2021-02-10 r79979)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C               LC_TIME=en_CA.UTF-8       
 [4] LC_COLLATE=en_CA.UTF-8     LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] testPkg_0.0.0.9000

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6                  XVector_0.31.1             
 [3] compiler_4.1.0              GenomeInfoDb_1.27.6        
 [5] zlibbioc_1.37.0             bitops_1.0-6               
 [7] prettyunits_1.1.1           MatrixGenerics_1.3.1       
 [9] tools_4.1.0                 testthat_3.0.2             
[11] digest_0.6.27               pkgbuild_1.2.0             
[13] pkgload_1.1.0               lattice_0.20-41            
[15] rlang_0.4.10                Matrix_1.3-2               
[17] DelayedArray_0.17.7         cli_2.3.0                  
[19] commonmark_1.7              parallel_4.1.0             
[21] xfun_0.21                   GenomeInfoDbData_1.2.4     
[23] withr_2.4.1                 stringr_1.4.0              
[25] roxygen2_7.1.1              xml2_1.3.2                 
[27] knitr_1.31                  desc_1.2.0                 
[29] S4Vectors_0.29.7            IRanges_2.25.6             
[31] grid_4.1.0                  stats4_4.1.0               
[33] rprojroot_2.0.2             Biobase_2.51.0             
[35] glue_1.4.2                  R6_2.5.0                   
[37] processx_3.4.5              callr_3.5.1                
[39] purrr_0.3.4                 magrittr_2.0.1             
[41] GenomicRanges_1.43.3        ps_1.5.0                   
[43] matrixStats_0.58.0          BiocGenerics_0.37.1        
[45] assertthat_0.2.1            SummarizedExperiment_1.21.1
[47] stringi_1.5.3               RCurl_1.98-1.2             
[49] crayon_1.4.1 
```

**Comment by ChristopherEeles** [comment]
I have confirmed the issue exists on Window as well. Same minimal reporduction as above.

Here is the Windows sessionInfo:
```
R Under development (unstable) (2021-02-15 r80013)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252    LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] PDATK_0.0.0.9000

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        roxygen2_7.1.1    prettyunits_1.1.1 ps_1.5.0          withr_2.4.1       crayon_1.4.1      rprojroot_2.0.2  
 [8] assertthat_0.2.1  R6_2.5.0          magrittr_2.0.1    cli_2.3.0         rlang_0.4.10      stringi_1.5.3     callr_3.5.1      
[15] testthat_3.0.2    xml2_1.3.2        desc_1.2.0        tools_4.1.0       stringr_1.4.0     glue_1.4.2        purrr_0.3.4      
[22] processx_3.4.5    xfun_0.21         pkgload_1.1.0     compiler_4.1.0    pkgbuild_1.2.0    knitr_1.31  
```

**Comment by RLumSK** [comment]
It appears that I can confirm this problem also for macOS. I just updated to the latest R-devel (my R-devel is a little bit older, because there is currently no newer pre-compiled version available) and I encountered the same problem. Contrary, the just released R 4.04 works fine. 

My session info:

```r
R Under development (unstable) (2021-02-07 r79964)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.1.0 tools_4.1.0   
```

**Comment by hadley** [comment]
Duplicate of https://github.com/r-lib/pkgload/issues/139
